demo_nodes_py/demo_nodes_py/topics/listener.py

```diff
@@ -36,9 +39,11 @@ def main(args=None):
         rclpy.spin(node)
     except KeyboardInterrupt:
         pass
-
-    node.destroy_node()
-    rclpy.shutdown()
+    except ExternalShutdownException:
+        sys.exit(1)
```

Comment:
Sorry to comment on an old ticket, but why are we exiting with error code 1? IIUC, `ExternalShutdownException` is something we should expect as part of the normal shutdown sequence here.

It seems to me that all of the demos should be updated to look like,

```py
try:
    rclpy.spin(node)
except (KeyboardInterrupt, ExternalShutdownException):
    pass
```

Reply:
See https://github.com/ros2/demos/pull/581